doc/sphinx/src/fortran.rst

```diff
@@ -0,0 +1,742 @@
+
+Noble-Abel
+^^^^^^^^^^
+
+.. code-block:: fortran
+
+   integer function init_sg_NobleAbel_f(matindex, eos, gm1, Cv, &
+                                      bb, qq, &
+                                      sg_mods_enabled, sg_mods_values) &
+     result(err)
+     integer(c_int), value, intent(in) :: matindex
+     type(sg_eos_ary_t), intent(in)    :: eos
+     real(kind=8), value, intent(in)   :: gm1, Cv, bb, qq
+     integer(kind=c_int), dimension(:), target, optional, intent(inout) :: sg_mods_enabled
+     real(kind=8), dimension(:), target, optional, intent(inout)        :: sg_mods_values
+   end function init_sg_NobleAbel_f
+
+Initializes a Noble-Abel EOS model at the specified material index.
+
+- ``matindex``: Material index (1-based in Fortran)
+- ``eos``: EOS array
+- ``gm1``: Gamma minus 1 (γ-1)
+- ``Cv``: Specific heat at constant volume
+- ``bb``: Covolume parameter
+- ``qq``: Energy shift parameter
+- ``sg_mods_enabled``: Optional array for enabling modifiers
+- ``sg_mods_values``: Optional array for modifier values
+
+Stiff Gas
+^^^^^^^^^
+
+.. code-block:: fortran
+
+   integer function init_sg_StiffGas_f(matindex, eos, gm1, Cv, &
+                                      Pinf, qq, &
+                                      sg_mods_enabled, sg_mods_values) &
+     result(err)
+     integer(c_int), value, intent(in) :: matindex
+     type(sg_eos_ary_t), intent(in)    :: eos
+     real(kind=8), value, intent(in)   :: gm1, Cv, Pinf, qq
+     integer(kind=c_int), dimension(:), target, optional, intent(inout) :: sg_mods_enabled
+     real(kind=8), dimension(:), target, optional, intent(inout)        :: sg_mods_values
+   end function init_sg_StiffGas_f
+
+Initializes a Stiff Gas EOS model at the specified material index.
+
+- ``matindex``: Material index (1-based in Fortran)
+- ``eos``: EOS array
+- ``gm1``: Gamma minus 1 (γ-1)
+- ``Cv``: Specific heat at constant volume
+- ``Pinf``: Pressure shift parameter
+- ``qq``: Energy shift parameter
+- ``sg_mods_enabled``: Optional array for enabling modifiers
+- ``sg_mods_values``: Optional array for modifier values
+
+SAP Polynomial
+^^^^^^^^^^^^^^
+
+.. code-block:: fortran
+
+   integer function init_sg_SAP_Polynomial_f(matindex, eos, rho0, a0, a1, a2c, &
+                                            a2e, a3, b0, b1, b2c, b2e, b3, &
+                                            sg_mods_enabled, sg_mods_values) &
+     result(err)
+     integer(c_int), value, intent(in) :: matindex
+     type(sg_eos_ary_t), intent(in)    :: eos
+     real(kind=8), value, intent(in)   :: rho0, a0, a1, a2c, a2e, a3,&
+              b0, b1, b2c, b2e, b3
+     integer(kind=c_int), dimension(:), target, optional, intent(inout) :: sg_mods_enabled
+     real(kind=8), dimension(:), target, optional, intent(inout)        :: sg_mods_values
+   end function init_sg_SAP_Polynomial_f
+
+Initializes a SAP (Separate Analytic Polynomials) Polynomial EOS model at the specified material index.
```

Comment:
I'm not sure this is what SAP stands for in this context 😉 . We should maybe consider renaming in the future, but suffice it to say that it's a specific implementation of _a_ polynomial EOS.